src/types/redis_bloom_chain.cc

```diff
@@ -147,18 +171,23 @@ rocksdb::Status BloomChain::Add(const Slice &user_key, const Slice &item, int *r
 
   // insert
   if (!exist) {
-    if (metadata.size + 1 > metadata.GetCapacity()) {  // TODO: scaling would be supported later
-      return rocksdb::Status::Aborted("filter is full");
+    if (metadata.size + 1 > metadata.GetCapacity()) {
+      if (metadata.IsScaling()) {
+        s = createBloomFilter(ns_key, &metadata);
+        if (!s.ok()) return s;
+      } else {
+        return rocksdb::Status::Aborted("filter is full and is nonscaling");
+      }
     }
     s = bloomAdd(bf_key_list.back(), item_string);
```

Comment:
Should we enlarge `bf_key_list` if scaling?

Reply:
Sure, it certainly needs to be enlarged. I think I mistakenly missed it.